File: Frontend/friend-nest/src/app/services/follow.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Follow } from '../models/Follow';
import { FollowerTo } from '../models/response/FollowerTo';
import { FollowerToRequest } from '../models/request/FollowerToRequest';

@Injectable({
  providedIn: 'root',
})
export class FollowService {
  constructor(private http: HttpClient) {}

  switchFollow(followingId: number, followedId: number): Observable<Follow> {
    const payload = {
      followingId: followingId,
      followedId: followedId,
    };
    return this.http.post<Follow>(
      'http://localhost:8080/api/switchFollow',
      payload,
    );
  }

  isFollowedByLoggedUser(
    followingId: number,
    followedId: number,
  ): Observable<boolean> {
    const payload = {
      followingId: followingId,
      followedId: followedId,
    };
    return this.http.post<boolean>(
      'http://localhost:8080/api/isFollowedByLoggedUser',
      payload,
    );
  }

  getFollowerList(
    followerToRequest: FollowerToRequest,
  ): Observable<FollowerTo[]> {
    return this.http.post<FollowerTo[]>(
      'http://localhost:8080/api/getFollowerList',
      followerToRequest,
    );
  }
}

File: Backend/friendnest/src/main/java/pl/polsl/friendnest/service/FollowService.java
package pl.polsl.friendnest.service;

import pl.polsl.friendnest.model.Follow;
import pl.polsl.friendnest.model.request.FollowerToRequest;
import pl.polsl.friendnest.model.response.FollowerTo;

import java.util.List;

public interface FollowService {

    Follow switchFollow(Long followingUserId, Long followedUserId);

    boolean isFollowedByLoggedUser(Long followingUserId, Long followedUserId);

    List<FollowerTo> getFollowerTo(FollowerToRequest followerToRequest);
}
